src/box/lua/session.c

```diff
@@ -51,17 +51,25 @@ static const char *sessionlib_name = "box.session";
 static int
 lbox_session_create(struct lua_State *L)
 {
+	int fd = luaL_optinteger(L, 1, -1);
+	enum session_type type = STR2ENUM(session_type,
+					  luaL_optstring(L, 2, "console"));
 	struct session *session = fiber_get_session(fiber());
 	if (session == NULL) {
 		session = session_create_on_demand();
 		if (session == NULL)
 			return luaT_error(L);
-		session->meta.fd = luaL_optinteger(L, 1, -1);
+		session->meta.fd = fd;
+		if (fd >= 0 && type != SESSION_TYPE_REPL) {
+			struct sockaddr_storage addrstorage;
+			struct sockaddr *addr = (struct sockaddr *)&addrstorage;
+			socklen_t addrlen = sizeof(addrstorage);
+			if (sio_getpeername(fd, addr, &addrlen) == 0)
```

Comment:
It might be better to use the already defined peer passed as a paramter in console.lua `local function client_handler(client, peer)`. The only other place is `local function start()` where it is a local descriptor anyway, no peer (unless it is a pipe though ...).

Reply:
Unfortunately, that parameter is a string, not a `sockaddr`. I considered storing a string instead of `sockaddr` in `session_meta`, but:
1. This looks less flexible: `sockaddr` can be printed in different formats and used for something else beside reporting peer name.
2. `string.format('%s:%s', peer.host, peer.port)` doesn't always yield the same result as `sio_strfaddr` - for Unix sockets the output may be different. I want the output to be the same for console and iproto connections and match `box.session.peer()`.

We could probably pass `sockaddr` in a hidden field of the `peer` table, but I don't want to complicate the patch at this point, because the benefit of it isn't clear. We can try to do it later though.

Reply:
Thanks for trying anyway.